inference-engine/src/gna_plugin/backend/gna_limitations.hpp

```diff
@@ -16,5 +17,63 @@ constexpr uint32_t convEachKernelByteAlignment = 16;
 constexpr uint32_t noOfInputsDivisor = 8;
 constexpr uint32_t noOfInputsLowPrecDivisor = 16;
 
-}
+namespace Cnn2D {
+struct RangeLimit {
+    uint32_t min;
+    uint32_t max;
+    std::string what;
+    bool isValid(const uint32_t val) const;
+    std::string GetErrorOrEmpty(const uint32_t val) const;
+};
+
+struct RangeLimit2D {
+    RangeLimit hLimit;
+    RangeLimit wLimit;
+    bool isValid(const uint32_t h, const uint32_t w) const;
+    std::string GetErrorOrEmpty(const uint32_t h, const uint32_t w) const;
+};
+
+struct RangeMultipleLimit : public RangeLimit {
+    uint32_t multiplier;
+    RangeMultipleLimit(RangeLimit rlIn, uint32_t multiplierIn);
+    bool isValid(const uint32_t val) const;
+    std::string GetErrorOrEmpty(const uint32_t val) const;
+};
+
+struct VectorOrSquareLimit {
+    uint32_t maxSquare;
+    uint32_t maxVectorHeight;
+    uint32_t maxVectorWidth;
+    bool isValid(const uint32_t h, const uint32_t w) const;
+    std::string GetErrorOrEmpty(const uint32_t h, const uint32_t w, std::string what) const;
+};
+
+struct VectorOrSquareLimitByChannels {
+    unsigned channelLimit;
+    VectorOrSquareLimit smallChannel;
+    VectorOrSquareLimit bigChannel;
+};
```

Comment:
Do we need `isValid` and `GetErrorOrEmpty` here?

Reply:
functions added